mne/minimum_norm/time_frequency.py

```diff
@@ -447,94 +452,125 @@ def compute_source_psd(raw, inverse_operator, lambda2=1. / 9., method="dSPM",
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
 
-        if is_free_ori and pick_ori is None:
-            sol = combine_xyz(sol, square=True)
-        else:
-            sol = (sol * sol.conj()).real
+    Returns
+    -------
+    stc : SourceEstimate | VolSourceEstimate
+        The PSD (in dB) of each of the sources.
 
-        if method != "MNE":
-            sol *= noise_norm ** 2
+    See Also
+    --------
+    compute_source_psd_epochs
 
-        psd += sol
-        n_windows += 1
+    Notes
+    -----
+    Each window is multiplied by a window before processing, so
+    using a non-zero overlap is recommended.
 
-    psd /= n_windows
+    This function is different from :func:`compute_source_psd_epochs` in that:
 
-    psd = 10 * np.log10(psd)
+    1. ``dB=True`` by default (deprecated; will change to False in 0.18)
+    2. ``bandwidth='hann'`` by default, skipping multitaper estimation
+    3. For convenience it wraps
+       :func:`mne.make_fixed_length_events` and :class:`mne.Epochs`.
 
-    subject = _subject_from_inverse(inverse_operator)
-    stc = _make_stc(psd, vertices=vertno, tmin=fmin * 1e-3,
-                    tstep=fstep * 1e-3, subject=subject,
-                    src_type=inverse_operator['src'].kind)
-    return stc
+    Otherwise the two should produce identical results.
+    """
+    if dB is None:
+        dB = True
+        warn('dB=True by default in 0.17 but will change to False in 0.18, '
+             'set it explicitly to avoid this warning', DeprecationWarning)
+    tmin = 0. if tmin is None else float(tmin)
+    overlap = float(overlap)
+    if not 0 <= overlap < 1:
+        raise ValueError('Overlap must be at least 0 and less than 1, got %s'
+                         % (overlap,))
+    n_fft = int(n_fft)
+    duration = ((1. - overlap) * n_fft) / raw.info['sfreq']
+    events = make_fixed_length_events(raw, 1, tmin, tmax, duration)
+    epochs = Epochs(raw, events, 1, 0, (n_fft - 1) / raw.info['sfreq'])
+    out = compute_source_psd_epochs(
+        epochs, inverse_operator, lambda2, method, fmin, fmax,
+        pick_ori, label, nave, pca, inv_split, bandwidth, adaptive, low_bias,
+        True, n_jobs, prepared, method_params, return_sensor=True)
+    source_data = 0.
+    sensor_data = 0.
+    count = 0
+    for stc, evoked in out:
+        source_data += stc.data
+        sensor_data += evoked.data
+        count += 1
+    assert count > 0  # should be guaranteed by make_fixed_length_events
+    sensor_data /= count
+    source_data /= count
+    if dB:
+        np.log10(sensor_data, out=sensor_data)
+        sensor_data *= 10.
+        np.log10(source_data, out=source_data)
+        source_data *= 10.
+    evoked.data = sensor_data
+    evoked.nave = count
+    stc.data = source_data
+    out = stc
+    if return_sensor:
+        out = (out, evoked)
```

Comment:
Fix: make `compute_source_psd` wrap to `compute_source_psd_epochs`. We get multitaper estimation for free, and simplify the codebase.

Enhancement: have `return_sensor` output so you get the Evoked and SourceEstimate.